src/main/scala/org/constellation/domain/healthcheck/HealthCheckConsensus.scala

```diff
@@ -0,0 +1,569 @@
+                  (initiallyUnknownPeers -- roundPeers.keySet)
+                    .map(
+                      _ -> RoundData(ProxyPeers(NonEmptySet.one(peerData)).asLeft[PeerData])
+                    )
+                val updatedProxyPeers =
+                  roundPeers
+                    .filterKeys(initiallyUnknownPeers.contains)
+                    .mapValues(
+                      roundData =>
+                        roundData.copy(
+                          peerData = roundData.peerData
+                            .leftMap(proxyPeers => proxyPeers.copy(proxyPeers.peersData.add(peerData)))
+                        )
+                    )
+
+                roundPeers ++ updatedProxyPeers ++ addedPeers
+            }
+
+            updatedPeers + (healthProposal.checkingPeerId -> prd)
+          }.getOrElse(roundPeers)
+        (updated, updated)
+      }
+    } yield ()
+
+  private def fetchProposals(): F[Unit] =
+    for {
+      isAcquired <- fetchingProposalLock.tryAcquire
+      _ <- if (isAcquired)
+        fetchHealthProposals()
+          .handleErrorWith(
+            e =>
+              fetchingProposalLock.release >> logger
+                .error(e)("Error fetching healthcheck proposal.") /*>> e.raiseError[F, Unit]*/
+          ) //not needed I think because every fetch failure is handled in fetchHealthProposals, should do the same for sendHealthProposal I think
+      else
+        F.unit
+      _ <- fetchingProposalLock.release
+    } yield ()
+
+  private def fetchHealthProposals(): F[Unit] =
+    for {
+      peers <- roundPeers.get
+        .map(_.filter { case (_, rd) => rd.consensusData.isEmpty })
+      roundIds <- roundIds.get
+      responses <- peers.toList.traverse {
+        case (id, roundData) => {
+          {
+            roundData.peerData match {
+              case Left(proxyPeers) =>
+                for {
+                  proxyPeer <- randomlyPickProxyPeer(proxyPeers.peersData)
+                  response <- PeerResponse.run(
+                    apiClient.healthcheck.fetchPeerHealthStatus(FetchPeerHealthStatus(roundIds.toSortedSet, id.some)),
+                    unboundedBlocker
+                  )(proxyPeer.peerMetadata.toPeerClientMetadata.copy(port = "9003"))
+                } yield response
+              case Right(peerData) =>
+                PeerResponse.run(
+                  apiClient.healthcheck.fetchPeerHealthStatus(FetchPeerHealthStatus(roundIds.toSortedSet)),
+                  unboundedBlocker
+                )(peerData.peerMetadata.toPeerClientMetadata.copy(port = "9003"))
+            }
+          }.handleErrorWith(
+            _ => FetchingFailed(id).asInstanceOf[FetchProposalError].asLeft[SendPerceivedHealthStatus].pure[F]
+          )
+        }.map(_.bimap(id -> _, id -> _))
+      }
+      (failed, succeeded) = responses.separate
+      _ <- succeeded.traverse { case (_, proposal) => processProposal(proposal.consensusHealthStatus) }
```

Comment:
if `processProposal` will throw an error, whole traverse will fail and further proposals won't be processed

Reply:
Right, will look into this.